app/validators/gov_uk_url_format_validator.rb

```diff
@@ -1,14 +1,25 @@
 # Accepts options[:message] and options[:allowed_protocols]
 class GovUkUrlFormatValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
-    unless %r{\A#{Whitehall.public_protocol}://#{Whitehall.public_host}/}.match?(value)
+    unless GovUkUrlFormatValidator.matches_gov_uk?(value) || matches_allow_list?(value)
```

Comment:
I think we can remove the `GovUkUrlFormatValidator` bit, as we're already inside the class.

Reply:
Unfortunately not.  I get a `NoMethodError: undefined method `matches_gov_uk?' for #<GovUkUrlFormatValidator:0x0000562d6d9863e0>` error without the class name.

Reply:
Ah, apologies. Hmm, how about `self.matches_gov_uk?`?

Reply:
It's the same error in that case too

Reply:
Looks like the [approach](https://stackoverflow.com/questions/2527830/ruby-calling-class-method-from-instance) is `self.class.matches_govuk_uk?`. Shame, not as terse as I thought it would be, but probably slightly better than repeating the class name, as:

1. it is quicker to establish that it's referring to itself (using the class name looks like we're calling out to another class, at first glance)
2. it's one less thing to replace if we ever rename the class

Very nitpicky of me though 😄

Reply:
That's a good point.  I've updated to use `self.class`.